90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -977,10 +977,19 @@ populate_be_list() {
       # All other datasets are ignored
       continue
     fi
+    if [ -n "${BOOTFS}" ] && [ "${BOOTFS}" = "${fs}" ] ; then
+      # If BOOTFS is defined, we'll manually append it to the array
+      continue
+    fi
+
     candidates+=( "${fs}" )
   done <<< "$(zfs list -H -o name,mountpoint,org.zfsbootmenu:active)"
```

Comment:
What about
```
$(zfs list -H -o name,mountoint,org.zfsbootmenu:active | sort -r)
```
to ensure `candidates` is presorted? This would make the `readarray` in Line 989 and the entire `sorted` array unnecessary.